### PR TITLE
Preprocessing refactor

### PR DESCRIPTION
**Pull request by lgraesser**

### CODE REVIEW

**Comment by kengz** [comment]
^ CI breaks only due to dqn and dummy not having implemented `recompile_model`. Straightforward fix. Disregard Codacy now, merge as needed.

**Comment by kengz** [comment]
Did some refactoring and renaming by convention. Still wish to do mroe refactoring on the top level later today

**Comment by lgraesser** [comment]
Checked dev_dqn for blowups and lunar_landing to benchmark performance vs. master. Able to replicate solve in 222 epis with current best params. But some catastrophically bad episodes also. Stability of param setting to be investigated.

File: rl/agent/dqn.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(self, env_spec,
                  gamma=0.95, learning_rate=0.1,
                  batch_size=16, n_epoch=5, hidden_layers_shape=[4],
                  hidden_layers_activation='sigmoid',
-                 output_layer_activation=None,
+                 output_layer_activation='linear',
```
Comment (kengz) [inline rl/agent/dqn.py]: potential breaker

**Pull request merged**